test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -193,28 +359,54 @@ public void WhenMoreThanOneSolutionExistsInTheDirectoryItPrintsErrorAndUsage()
-                .Where((p) => p.Name == "Lib")
-                .ToList();
+            var slnPath = Path.Combine(projectDirectory, "App.sln");
 
-            matchingProjects.Count.Should().Be(1);
-            var slnProject = matchingProjects[0];
-            slnProject.FilePath.Should().Be(projectPath);
-            slnProject.TypeGuid.Should().Be(ProjectTypeGuids.CPSProjectTypeGuid);
-            if (!string.IsNullOrEmpty(projectGuid))
+            if (string.IsNullOrEmpty(expectedProjectGuid))
             {
-                slnProject.Id.Should().Be(projectGuid);
+                var slnFile = SlnFile.Read(slnPath);
+                var matchingProjects = slnFile.Projects
+                    .Where((p) => p.Name == "Lib")
+                    .ToList();
+
+                matchingProjects.Count.Should().Be(1);
+                var slnProject = matchingProjects[0];
+                expectedProjectGuid = slnProject.Id;
             }
 
-            var restoreCmd = new DotnetCommand()
+            var expectedSlnContents = expectedSlnContentsTemplate.Replace("__PROJECTGUID__", expectedProjectGuid);
+            File.ReadAllText(slnPath)
+                .Should().Be(expectedSlnContents);
+
+            new DotnetCommand()
                 .WithWorkingDirectory(projectDirectory)
-                .Execute($"restore {Path.Combine("App", "App.csproj")}");
+                .Execute($"restore App.sln")
+                .Should().Pass();
 
-            var buildCmd = new DotnetCommand()
+            new DotnetCommand()
                 .WithWorkingDirectory(projectDirectory)
-                .Execute("build App.sln");
-            buildCmd.Should().Pass();
+                .Execute("build App.sln --configuration Release")
+                .Should().Pass();
+
+            var reasonString = "should be built in release mode, otherwise it means build configurations are missing from the sln file";
+
+            if (testAsset != "TestAppWithEmptySln")
+            {
+                var appReleaseDirectory = Directory.EnumerateDirectories(
+                    Path.Combine(projectDirectory, "App", "bin"),
+                    "Release",
+                    SearchOption.AllDirectories);
+                appReleaseDirectory.Count().Should().Be(1, $"App {reasonString}");
+                Directory.EnumerateFiles(appReleaseDirectory.Single(), "App.dll", SearchOption.AllDirectories)
+                    .Count().Should().Be(1, $"App {reasonString}");
+            }
+
+            var libReleaseDirectory = Directory.EnumerateDirectories(
+                Path.Combine(projectDirectory, "Lib", "bin"),
+                "Release",
+                SearchOption.AllDirectories);
+            libReleaseDirectory.Count().Should().Be(1, $"Lib {reasonString}");
+            Directory.EnumerateFiles(libReleaseDirectory.Single(), "Lib.dll", SearchOption.AllDirectories)
+                .Count().Should().Be(1, $"Lib {reasonString}");
```

Comment:
A test with numerous assertions is often better broken into several tests. This gives the opportunity for more granular, informative failures. 

The `if` block is especially indicative that something different is being tested in this case.